api/controllers/datasets.py

```diff
@@ -87,11 +87,11 @@ def delete(credentials, did):
 def create(credentials, tid, name):
     ensure_owner_or_admin(tid, credentials["id"])
 
-    if len(name) > 27 or not bool(re.search("^[a-zA-Z0-9_-]*$", name)):
+    if not bool(re.fullmatch("[a-zA-Z0-9]{1,62}", name)):
```

Comment:
I think we might need to make this a bit shorter as you suggested — when we submit BatchTransform jobs the regex we need to follow is `^[a-zA-Z0-9](-*[a-zA-Z0-9]){0,62}`, but batch transform job names can look something like: `ts1635886368-bertstyleqav3-fairness-squadv2-1637768686` i.e we have timestamp + model name + perturbation name (if any) + dataset name + number. We can't predict how large model names will be but:
- the ts is usually `ts` + 10 numbers + 1 dash => 13,
- `robustness` + 1 dash is the largest perturbation prefix => 11 (as you mentioned)
- 10 digit number at end => 10
Which leaves 62-34 = 28 spaces for model_name + 1 dash + dataset_name + 1 dash. So really 26 spaces for model_name + dataset_name. I think if we want to be safe we can probably restrict this one to `13`, and then have a separate issue in dynalab-cli to impose 13-length model names when uploading?

I think I might be missing something though, because I definitely see `20`-length dataset names on dynabench like `mnli-test-mismatched` evaluation for > 6-length model names. Maybe @TristanThrush knows?

Reply:
Apart from that, this PR LGTM!